frontend/src/components/NewRunParametersV2.tsx

```diff
@@ -57,6 +57,24 @@ interface NewRunParametersProps {
   handleParameterChange?: (parameters: RuntimeParameters) => void;
 }
 
+const inputConverter = (paramStr: string, paramTypeIdx: number) => {
+  if (ParameterType_ParameterTypeEnum[paramTypeIdx] === 'BOOLEAN' && paramStr === 'True') {
+    return true;
+  } else if (ParameterType_ParameterTypeEnum[paramTypeIdx] === 'BOOLEAN' && paramStr === 'False') {
+    return false;
+  } else if (
+    ParameterType_ParameterTypeEnum[paramTypeIdx] === 'NUMBER_INTEGER' &&
+    Number(paramStr)
+  ) {
+    return Number(paramStr);
+  } else if (ParameterType_ParameterTypeEnum[paramTypeIdx] === 'STRING') {
+    return paramStr;
+  } else {
+    // TODO: (jlyaoyuli) Determine which type (null or undefined) is returned for invalid input.
```

Comment:
`undefined` is a type, while `null` is an object. So if you are returning something, my best bet will be to return an object which is `null`. 

`undefined` means a variable has never been assigned any value.